File: NPC.cpp
/*
	author Sonny Theo Tumbur
*/

#include <iostream>
#include "NPC.h"

using namespace std;

NPC::NPC(){ //default ctor
	
}

NPC::NPC(const NPC& _npc){
	this->nama = _npc.nama;
	this->warnaNPC = _npc.warnaNPC;
//	this->MonsterNPC[] = _npc.MonsterNPC[]; //tidak bitwise copy karena bukan pointer
}

NPC::~NPC(){
	cout << "dtor NPC" << endl;
}

void NPC::talk() {
	cout << "Hello Player" << endl;
}

File: Android/model/NPC.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

import java.util.ArrayList;

/**
 *
 * @author Sonny Theo Thumbur
 */
public class NPC {
    private String nama; //identitas unik
    private ArrayList<Monster> monsterNPC;
    private String warnaNPC;
    
    public String getName() {
        return nama;
    }
    
    public ArrayList<Monster> getMonsterList() {
        return monsterNPC;
    }
    
    public Monster getFirstMonster() {
        return monsterNPC.get(0);
    }
    
    public String getWarna() {
        return warnaNPC;
    }
    
    public void talk() {
        //diletakan pada screen
    }
}
